src/citrine/resources/data_concepts.py

```diff
@@ -51,9 +56,10 @@ class DataConcepts(PolymorphicSerializable['DataConcepts']):
     Only populated if the :func:`get_type` method is invoked.
     """
 
-    def __init__(self, typ: str):
+    def __init__(self, typ: str, audit_info: Optional[Dict[str, Any]] = None):
```

Comment:
Right now `audit_info` is just a dictionary. It's up to the user to know that it might have keys `created_by`, etc. Should we create an actual `AuditInfo` class?

Reply:
I think that's the way to go, yeah.

Reply:
I've added the AuditInfo class, and the field `audit_info` now comes in through the constructor. I wasn't sure if this called for a leading underscore or not, but since we want it to be read-only, I think it does.

Reply:
> Maybe you can add a new class extending DataConcepts which has all this logic but is only extended by Data Objects and Templates
That might be a good idea, but at the moment the only thing it would get you is the `_client_keys` field. I don't think that on its own is worth the complexity of adding a new class.

Reply:
Well it also keeps `audit_info` out of other classes which that data doesn't apply to. Doesn't leaving it here mean `Property`s will also have an `audit_info` field?
> I've added the AuditInfo class, and the field audit_info now comes in through the constructor.
This is more functional but I actually preferred your old `setattr` approach since having it in the constructor sort of suggests that the user is allowed to set `audit_info` to whatever value they like.

Reply:
Those are both good points. The annoying thing about the `setattr` approach was that it did a blind set, so `self.audit_info` became a dictionary instead of an object. I think the right thing to do is to use setters, so this `Storable` object will create an `AuditInfo` object when you pass it  dictionary with `created_by`, `created_at`, etc.

Reply:
I made a `Storable` class that goes in between `DataConcepts` and all of the non-Attributes, mimicking the datasvc structure. This class has a setter for `audit_info` that can accept an `AuditInfo` class or a serialized dict version. `AuditInfo` is now a `Serializable` class and its fields are `Properties`.